crates/codegen/parser_runtime/src/napi/napi_parse_error.rs

```diff
@@ -0,0 +1,35 @@
+use napi_derive::napi;
+
+use super::*;
+use napi_text_index::*;
+
+#[napi(namespace = "parse_error")]
+#[derive(PartialEq, Clone)]
+pub struct ParseError(RustParseError);
+
```

Comment:
Since we now have a napi feature/gate, I suggest keeping every type/API in a single place. This makes it much easier to maintain or edit together, instead of editing one version and missing the other.

So instead of:

- `src/foo.rs` shared + rust stuff
- `src/bar.rs` shared + rust stuff

- `src/napi/foo.rs` napi stuff
- `src/napi/bar.rs` napi stuff

I suggest:

- `src/foo/mod.rs` shared stuff
- `src/foo/rust.rs` rust stuff
- `src/foo/napi.rs` napi stuff

- `src/bar/mod.rs` shared stuff
- `src/bar/rust.rs` rust stuff
- `src/bar/napi.rs` napi stuff

Reply:
FFI issues mean that we need the wrappers. But am I misunderstanding your suggestion? Is this just about changing the dir structure?

Reply:
I'm talking about the dir structure. For example, if I'm adding a new method to `Cursor`, it is really easy to miss its NAPI counterpart if it is not within the same module/folder. This is a trivial example, but there are more complicated scenarios.

The more atomic/holistic each type/module can be, the easier it is to update/revision accurately.

Reply:
Later PR. This was like this before this PR.

Reply:
Sounds good.

Reply:
#572